### PR TITLE
Mesh painting

### PR DESCRIPTION
**Pull request by jonathan2222**
**Mesh Painting**
- Added a MeshPainting Component which takes a mask texture as data.
- Changed draw args to also have extension data which can be set per entity. Right now it only allows for texture data, but buffers should be added later.
- Added a MeshPaint Custom Renderer to the render graphs.
- Changed the MeshKey to be different if its entity mask is different.

**Render Graph Editor**
- The user can now change the render stage dimensions after it has been created.
- Added options when saving a render graph.

**Game Console**
- Added an array type to ConsoleCommand as an argument.
- Made the PushInfo, PushError, and PushMsg public in the Game Console.

**Commands**
- Changed the command "debug_texture" to show multiple textures instead of one.

### CODE REVIEW

File: Assets/Shaders/DeferredGeometry/Geom.vert [hunk 1]
```diff
@@ -28,8 +29,10 @@ void main()
 	vec4 worldPosition      = instance.Transform * vec4(vertex.Position.xyz, 1.0f);
 	vec4 prevWorldPosition  = instance.PrevTransform * vec4(vertex.Position.xyz, 1.0f);
 
-	vec3 normal 	        = normalize((instance.Transform * vec4(vertex.Normal.xyz, 0.0f)).xyz);
-	vec3 tangent            = normalize((instance.Transform * vec4(vertex.Tangent.xyz, 0.0f)).xyz);
+	mat4 normalTransform = instance.Transform;//transpose(inverse(instance.Transform));
```
Comment (Mumsfilibaba) [inline Assets/Shaders/DeferredGeometry/Geom.vert]: Remove commented code, or is it necessary?

**Review by Mumsfilibaba — CHANGES_REQUESTED** [review]
Look at the linter warnings. Should be const reference as the linter says 👀

File: Assets/Shaders/DeferredGeometry/Geom.vert (hunk 1 above)
Comment (jonathan2222) [reply Assets/Shaders/DeferredGeometry/Geom.vert]: ok, will do.

Comment (Ceanze) [reply Assets/Shaders/DeferredGeometry/Geom.vert]: It is fixed 🤠

**Review by Mumsfilibaba — APPROVED** [review]
Very nice 👍

File: CrazyCanvas/Source/CrazyCanvas.cpp [hunk 2]
```diff
@@ -135,5 +135,22 @@ bool CrazyCanvas::LoadRendererResources()
...
+		Sampler* pNearestSampler = Sampler::GetNearestSampler();
+
+		ResourceUpdateDesc cubeTextureUpdateDesc = {};
+		cubeTextureUpdateDesc.ResourceName = "BRUSH_MASK";
+		cubeTextureUpdateDesc.ExternalTextureUpdate.ppTextures = &pTexture;
+		cubeTextureUpdateDesc.ExternalTextureUpdate.ppTextureViews = &pTextureView;
+		cubeTextureUpdateDesc.ExternalTextureUpdate.ppSamplers = &pNearestSampler;
+
+		RenderSystem::GetInstance().GetRenderGraph()->UpdateResource(&cubeTextureUpdateDesc);
+	}
```
Comment (norda97) [inline CrazyCanvas/Source/CrazyCanvas.cpp:169]: Can be moved to mesh painter custom renderer for clarity over ownership

**Review by norda97 — APPROVED** [review]
Good job! Look at my comment ;)

File: CrazyCanvas/Source/CrazyCanvas.cpp (hunk 2 above)
Comment (Ceanze) [reply CrazyCanvas/Source/CrazyCanvas.cpp:169]: Creating external resources in either Init or RenderGraphInit is not a working feature of the current implementation. Therefore it is here :)

**Pull request merged**